### PR TITLE
MBS-8905, MBS-8927, MBS-9120: Allow for more auto-edits

### PR DESCRIPTION
**Pull request by y-van-z**
Allow for auto-edits in the three following cases:
- Editing release in a 1 hour delay, see [MBS-8905](https://tickets.metabrainz.org/browse/MBS-8905)
- Lowering release data quality in a 1 hour delay, see [MBS-8927](https://tickets.metabrainz.org/browse/MBS-8927)
- Adding time and setlist to an event that doesn’t have one, see [MBS-9120](https://tickets.metabrainz.org/browse/MBS-9120)

Although there is no test for amending release for now, this change doesn’t conceal other tests.

### CODE REVIEW

**Comment by y-van-z** [comment]
Note: Technically `AllowAmendingRelease` should not be a `Role` anymore, but I want to add parameters (that specify edit table name and create edit type) to it for [MBS-8134](https://tickets.metabrainz.org/browse/MBS-8134).

File: lib/MusicBrainz/Server/Edit/Release/ChangeQuality.pm [hunk 1]
```diff
@@ -88,5 +89,11 @@ method accept
     );
 }
 
+sub allow_auto_edit
+{
+    my $self = shift;
+    return $self->can_amend && $self->data->{new}{quality} != $QUALITY_HIGH;
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Edit/Release/ChangeQuality.pm]: Reversing these will save a query against the edit table if the quality is high.

**Comment by mwiencek** [comment]
I'm surprised that the calling order is undefined for multiple `around` modifiers. [For superclasses it seems to be defined](http://search.cpan.org/~ether/Moose-2.2004/lib/Moose/Manual/MethodModifiers.pod#Execution_order_of_method_modifiers_and_inheritance), and I assumed for roles it depended on where the `with` statement is inserted, but guess not.

But your solution is much more straightforward, so that's good.

**Comment by y-van-z** [comment]
You are most probably correct about this, although the documentation is incomplete.
Anyway, it would not be very handy and we need it as a method to change release quality.

File: lib/MusicBrainz/Server/Edit/Release/ChangeQuality.pm (hunk 1 above)
Comment (y-van-z) [inline lib/MusicBrainz/Server/Edit/Release/ChangeQuality.pm]: Done in https://github.com/metabrainz/musicbrainz-server/pull/479/commits/3b899da260a9de1e83a6f9257f4008615f813cb3.

**Pull request merged**